File: app/src/main/java/me/magnum/melonds/model/RendererConfiguration.java
package me.magnum.melonds.model;

public class RendererConfiguration {
	private VideoFiltering videoFiltering;

	public RendererConfiguration(VideoFiltering videoFiltering) {
		this.videoFiltering = videoFiltering;
	}

	public VideoFiltering getVideoFiltering() {
		return videoFiltering;
	}
}

File: app/src/main/java/me/magnum/melonds/domain/model/RendererConfiguration.kt
package me.magnum.melonds.domain.model

import me.magnum.melonds.domain.model.render.RenderStrategy

data class RendererConfiguration(
    val renderer: VideoRenderer,
    val videoFiltering: VideoFiltering,
    val threadedRendering: Boolean,
    val renderStrategy: RenderStrategy,
    private val internalResolutionScaling: Int,
) {

    val resolutionScaling get() = when (renderer) {
        VideoRenderer.SOFTWARE -> 1
        VideoRenderer.OPENGL -> internalResolutionScaling
        VideoRenderer.COMPUTE -> internalResolutionScaling
    }
}